libraries/eosiolib/contracts/eosio/privileged.hpp

```diff
@@ -185,6 +212,217 @@ namespace eosio {
     */
    void get_blockchain_parameters(eosio::blockchain_parameters& params);
 
+   /**
+    *  Helper structure for host functions set_parameters_packed and get_parameters_packed
+    *  @ingroup privileged
+    */
+   struct params_object{
+      std::string packed;
+
+      params_object() = default;
+
+      template<typename T>
+      params_object(T v){
+         (*this)(v);
+      }
+
+      params_object(const std::string& params) : packed(params) {}
+
+      bool operator == (const params_object& pp) const{
+         return packed == pp.packed;
+      }
+
+      params_object& operator () (unsigned_int val){
+         char buffer[4];
+         datastream<char*> ds((char*)&buffer, sizeof(buffer));
+         ds << val;
+         ASSERT_LE(ds.tellp(), sizeof(buffer));
+
+         packed += {buffer, ds.tellp()};
+         return *this;
+      }
+
+      params_object& operator () (uint16_t val){
+         char buffer[2];
+         datastream<char*> ds((char*)&buffer, sizeof(buffer));
+         ds << val;
+         ASSERT_EQ(ds.tellp(), sizeof(buffer));
+
+         packed += {buffer, ds.tellp()};
+         return *this;
+      }
+
+      params_object& operator () (uint32_t val){
+         char buffer[4];
+         datastream<char*> ds((char*)&buffer, sizeof(buffer));
+         ds << val;
+         ASSERT_EQ(ds.tellp(), sizeof(buffer));
+
+         packed += {buffer, ds.tellp()};
+         return *this;
+      }
+
+      params_object& operator () (uint64_t val){
+         char buffer[8];
+         datastream<char*> ds((char*)&buffer, sizeof(buffer));
+         ds << val;
+         ASSERT_EQ(ds.tellp(), sizeof(buffer));
+
+         packed += {buffer, ds.tellp()};
+         return *this;
+      }
+
+      params_object& operator () (std::vector<unsigned_int>&& val){
+         char buffer[512];
+         datastream<char*> ds((char*)&buffer, sizeof(buffer));
+         ds << val;
+         ASSERT_LESS(ds.tellp(), sizeof(buffer));
+
+         packed += {buffer, ds.tellp()};
+         return *this;
+      }
+
+   };
+
+    /**
+     * Set the blockchain parameters (version using the set_parameters_packed host function)
+     *
+     * @ingroup privileged
+     *
+     * @param New blockchain parameters to set
+    */
+    inline void set_parameters(params_object& params) {
```

Comment:
This should be `const param_objects&`.  It's small but it can help the compiler to optimize a bit better as it doesn't know it the host function will mutate data or not.

Reply:
I removed this code, as the get_parameters_packed and set_parameters packed host functions were implemented in a different way.